src/thumbnails/doc.md

```diff
@@ -0,0 +1,210 @@
+
+```
+moov.trak.mdia.minf.stbl.stss
+```
+- **Purpose**: Sync sample indices (I-frames only)
+- **Structure**: Array of keyframe sample numbers
+- **Usage**: Target only I-frames for thumbnails
+
+```
+moov.trak.mdia.minf.stbl.stts
+```
+- **Purpose**: Sample timing and duration
+- **Structure**: Array of {sample_count, sample_delta}
+- **Usage**: Calculate precise timestamps
+
+```
+moov.trak.mdia.minf.stbl.stsz
+```
+- **Purpose**: Sample sizes for byte range calculation
+- **Structure**: Array of sample sizes
+- **Usage**: Determine exact byte ranges to download
+
+```
+moov.trak.mdia.minf.stbl.stsc
+```
+- **Purpose**: Map samples to chunks
+- **Structure**: Array of {first_chunk, samples_per_chunk, sample_description_index}
+- **Usage**: Calculate chunk boundaries
+
+```
+moov.trak.mdia.minf.stbl.stco/co64
+```
+- **Purpose**: Chunk offset locations in file
+- **stco**: 32-bit offsets
+- **co64**: 64-bit offsets
+- **Usage**: Calculate absolute file positions
+
+## Extraction Process
+
+### 1. Format Detection
+- Skip MP3 files (no video)
+- Support MP4 family formats 
+- Proceed with extraction for compatible formats
+
+### 2. Video Track Analysis
+- Locate `moov` box efficiently (8KB start/end search)
+- Navigate `moov → trak → mdia → minf → stbl` hierarchy
+- Identify video tracks via `hdlr` box (handler type "vide")
+
+### 3. Sample Table Processing
+- **STSS**: Keyframe indices for I-frame targeting
+- **STTS**: Sample timing for timestamp calculation
+- **STSZ**: Sample sizes for byte range calculation
+- **STSC**: Sample-to-chunk mapping
+- **STCO/CO64**: Chunk offset locations
+- **STSD**: AVCC configuration (SPS/PPS)
+
+### 4. Target Sample Selection
+- Prefer I-frames from STSS entries (5-10% of total samples)
+- Distribute evenly across available keyframes
+- Fallback to regular samples if no I-frame info
+
+### 5. Optimized Download
+- Calculate precise byte ranges for target samples
+- Download only necessary sample data
+- Extract parameter sets from AVCC or inline samples
+
+### 6. H.264 Decoding
+- Initialize OpenH264 decoder with SPS/PPS
+- Extract NALUs from MP4 samples (length-prefixed)
+- Decode IDR frames to YUV420p
+- Convert YUV→RGB→JPEG with quality=85
+
+## H.264 Processing Pipeline
+
+### Sample Structure
+- **Format**: Length-prefixed NAL units (4-byte length + NAL data)
+- **NAL Types**: SPS(7), PPS(8), IDR(5), Non-IDR(1)
+- **Parameter Sets**: Extract from `stbl.stsd.avc1.avcC` or inline samples
+
+### Decoding Flow
+1. **Initialize**: OpenH264 decoder with SPS/PPS parameter sets
+2. **Extract**: NAL units from MP4 samples (length-prefixed format)
+3. **Convert**: To Annex-B format (0x00000001 start codes)
+4. **Decode**: IDR frames to YUV420p using OpenH264
+5. **Convert**: YUV→RGB→JPEG with quality=85 and resizing
+
+## Usage Examples
+
+### Local File Extraction
+```rust
+use mediaparser::thumbnails::extract_thumbnails;
+
+let thumbnails = extract_thumbnails("video.mp4", 5, 320, 240).await;
```

Comment:
Should we allow for specifying what video track to extract the thumbnails from? It looks like we default to the first one we find. Probably the most flexible API would be allowing the caller to pass in a callback function that takes in a list of available tracks and returns the index of the track to use for thumbnails.

Perhaps that could be an enhancement for another PR. If we switch to accepting a config object rather than individual parameters, it'd be easy to add a new property that supports that.